lib/usb_connection.js

```diff
@@ -189,12 +190,10 @@ USB.Connection.prototype.end = function() {
     // Tell the USB daemon to end all processes active
     // on account of this connection
     Daemon.deregister(self, function(err) {
-      self._close();
       if (err) {
-        reject(err);
-      } else {
-        resolve();
+        return reject(err);
       }
+      self._close(resolve);
```

Comment:
This was very bad. We need to wait for the close to complete before continuing...